clients/da-vinci-client/src/main/java/com/linkedin/davinci/store/rocksdb/RocksDBStorageEngineFactory.java

```diff
@@ -139,6 +140,15 @@ public RocksDBStorageEngineFactory(
     this.env.setBackgroundThreads(rocksDBServerConfig.getRocksDBEnvFlushPoolSize(), Priority.HIGH);
     this.env.setBackgroundThreads(rocksDBServerConfig.getRocksDBEnvCompactionPoolSize(), Priority.LOW);
 
+    long cacheBytesNeeded =
+        rocksDBServerConfig.getRocksDBBlockCacheSizeInBytes() + (rocksDBServerConfig.isUseSeparateRMDCacheEnabled()
+            ? rocksDBServerConfig.getRocksDBRMDBlockCacheSizeInBytes()
+            : 0);
+    if (Runtime.getRuntime().maxMemory() * 0.8 < cacheBytesNeeded) {
```

Comment:
can we make this threshold configurable?
Also can multiple rocksDB be instantiated resulting in a combined usage of more than the allowed memory?

Reply:
I don't think this will work.  The Runtime will tell you information about the JVM, but the rocksdb data usage won't be billed against that?

Reply:
blockcache is shared amongst all rocksdb engines.

Reply:
Yeah, Zac is correct.
One way in Oracle JVM (getting this from stackoverflow):
```
        long memorySize = ((com.sun.management.OperatingSystemMXBean) ManagementFactory.getOperatingSystemMXBean()).getTotalPhysicalMemorySize();
```
Not sure whether it works with OpenJDK or not.